### PR TITLE
Stream::all implementation

### PR DESCRIPTION
**Pull request by shekohex**
- [x] Adds stream::all implementation.
- [x] update method docs
- [x] an example for stream `all` method.
- [x] tests

### CODE REVIEW

File: src/stream/stream.rs [hunk 1]
```diff
@@ -168,3 +231,50 @@ impl<S: futures::Stream> futures::Stream for Take<S> {
...
+
+#[derive(Debug)]
+pub struct AllFuture<'a, S, F, T>
+where
+    F: FnMut(T) -> bool,
+{
+    stream: &'a mut S,
+    f: F,
+    result: bool,
+    __item: PhantomData<T>,
```
Comment (shekohex) [inline src/stream/stream.rs]: I don't know is that a good way to name the `PhantomData` fields, feel free to comment on this!

Comment (yoshuawuyts) [reply src/stream/stream.rs]: This looks fine!

File: src/stream/stream.rs [hunk 2]
```diff
@@ -111,6 +112,68 @@ pub trait Stream {
...
+    /// let mut s = stream::empty::<u32>();
+    /// assert!(s.all(|_| false).await);
+    ///
+    /// #
+    /// # }) }
+    /// ```
+    /// Stopping at the first `false`:
+    ///
+    /// TODO: add example here
+    /// TODO: add more examples
```
Comment (yoshuawuyts) [inline src/stream/stream.rs]: I think it's fine to just have the examples you already made!

File: src/stream/stream.rs [hunk 3]
```diff
@@ -111,6 +112,68 @@ pub trait Stream {
...
+    ///
+    /// ```
+    /// # fn main() { async_std::task::block_on(async {
+    /// #
+    /// use async_std::prelude::*;
+    /// use async_std::stream;
+    ///
+    /// let mut s = stream::empty::<u32>();
+    /// assert!(s.all(|_| false).await);
+    ///
```
Comment (yoshuawuyts) [inline src/stream/stream.rs]: Nit: extra newline here.
```suggestion
```

File: src/stream/stream.rs (hunk 2 above)
Comment (shekohex) [reply src/stream/stream.rs]: ok

**Comment by jamesmunns** [comment]
@yoshuawuyts @shekohex is this ready to merge?

**Comment by shekohex** [comment]
@jamesmunns yes!
seems ok to me

**Pull request merged**